pkg/cluster/context.go

```diff
@@ -349,6 +351,17 @@ func (cc *createContext) provisionControlPlane(
 		}
 	}
 
+	// Wait for control plane node to reach Ready status.
+	if cc.waitForControlPlane {
+		if !node.WaitForControlPlane(time.Now().Add(time.Second * 30)) {
+			// TODO: logging here
+			if !cc.retain {
+				nodes.Delete(*node)
+			}
+			return "", fmt.Errorf("timed out waiting for control plane to be ready")
```

Comment:
it should not return "", because the kubeadmConfig still needs to be cleaned up afterwards.

but also, let's just log a warning here.